### PR TITLE
Separate last year's news

### PR DESCRIPTION
**Pull request by gernotheiser**
Signed-off-by: Gernot Heiser <gernot@unsw.edu.au>

### CODE REVIEW

File: content/news.pml [hunk 1]
```diff
@@ -1159,293 +1159,8 @@
...
-
-   More information on the <a
-   href="https://microkerneldude.wordpress.com/2020/04/07/the-sel4-foundation-what-and-why/">Blog
-   post</a> about "The seL4 Foundation &mdash; What and Why", and in
-   the press releases from <a
-   href="https://www.linuxfoundation.org/press-release/2020/04/sel4-microkernel-optimized-for-security-gets-support-of-linux-foundation/">the
-   Linux Foundation</a> and from <a
-   href="https://www.csiro.au/en/News/News-releases/2020/seL4-developers-create-open-source-foundation">CSIRO's
-   Data61</a>.
+    Older news: <a href="news-20.pml">2020</a>
```
Comment (axel-h) [inline content/news.pml:1164]: Putting them to a separate page (and repeating this every year) will break the URLs that I try to add in the other PR for each news items. Now one could argue that if we do this in September each year most people wont be affected, but I would not rule out that eventually, older news item also will be referenced, and a permanent URL would be nice. At some point, there might be no way to handle this with just static web pages (without JavaScript and AJAX concepts) and a CMS becomes necessary.

Comment (gernotheiser) [reply content/news.pml:1164]: I agree that it's bad to change URLs, but that's not hard to achieve

**Comment by gernotheiser** [comment]
Ok, the drawback is that there's no longer a generic news page to reference. Can this be done with a redirect @wom-bat?

**Comment by lsf37** [comment]
> Ok, the drawback is that there's no longer a generic news page to reference. Can this be done with a redirect @wom-bat?

Another way would be to have a scheme, eg. `news-2021.html`, `news-2020.html` etc, and only change which one the front page links to. That would leave all old URLs permanent (once we have switched to the scheme).

**Comment by lsf37** [comment]
> Another way would be to have a scheme, eg. news-2021.html, news-2020.html etc, and only change which one the front page links to. That would leave all old URLs permanent (once we have switched to the scheme).

Sorry, just saw that this is what you are already doing.

> Ok, the drawback is that there's no longer a generic news page to reference. Can this be done with a redirect @wom-bat?

Yes, a redirect would work for that, could also be symbolic link, and the links suggested in the page would reference the version with date instead.

**Comment by gernotheiser** [comment]
Didn't know you could use symlinks. But do they have the right behaviour? The desired behaviour is that when you click on the generic news link, you should get to the page with the year identification, so that if you take a link from there it should remain unchanged moving into the next year.

**Comment by lsf37** [comment]
> Didn't know you could use symlinks. But do they have the right behaviour? The desired behaviour is that when you click on the generic news link, you should get to the page with the year identification, so that if you take a link from there it should remain unchanged moving into the next year.

True, a redirect would be the better choice for this one.

**Comment by gernotheiser** [comment]
@wom-bat ?

File: content/news.pml (hunk 1 above)
Comment (wom-bat) [reply content/news.pml]: Simplest fix is to make a directory 'news', name the newsfiles 'news/20.pml', 'news/21.pml' etc; and make 'news.pml' just redirect to the latest.  That way, you never have to alter URLs.

**Comment by gernotheiser** [comment]
Superseded by #140

**Pull request closed without merging**